vmdk_plugin/plugin.go

```diff
@@ -117,19 +120,70 @@ func (d *vmdkDriver) mountVolume(name string) (string, error) {
 		return mountpoint, err
 	}
 
+	if d.useMockEsx {
+		return mountpoint, fmt.Errorf("No device to mount.")
+	}
+
+	skipInotify := false
+
+	watcher, err := inotify.NewWatcher()
+
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name, "dir": mountpoint},
+		).Error("Failed to create watcher ")
+		skipInotify = true
+	} else {
+		err = watcher.Watch(watchPath)
+		if err != nil {
+			log.WithFields(
+				log.Fields{"name": name, "dir": mountpoint},
+			).Error("Failed to watch /dev ")
+			skipInotify = true
+		}
+	}
+
 	// Have ESX attach the disk
 	dev, err := d.ops.Attach(name, nil)
 	if err != nil {
 		return mountpoint, err
 	}
 
-	time.Sleep(3 * time.Second)
+	device, err := fs.GetDevicePath(dev)
+	if err != nil {
+		return mountpoint, err
+	}
 
-	if d.useMockEsx {
-		return mountpoint, fs.Mount(mountpoint, nil, "ext4")
+	if skipInotify {
+		time.Sleep(sleepBeforeMount * time.Second)
+		return mountpoint, fs.Mount(mountpoint, "ext2", device)
```

Comment:
yes.. cause we cannot depend on inotify, it is better than failing a valid mount.